src/storage/redis_db.cc

```diff
@@ -777,4 +782,208 @@ rocksdb::Status Database::Copy(const std::string &key, const std::string &new_ke
+          list_db.Range(key, -count - offset, -1 - offset, &str_vec);
+          std::reverse(str_vec.begin(), str_vec.end());
+        } else {
+          list_db.Range(key, offset, offset + count - 1, &str_vec);
+        }
+      } else {
+        list_db.Range(key, 0, -1, &str_vec);
+      }
+    } else if (type == RedisType::kRedisSet) {
+      auto set_db = redis::Set(storage_, namespace_);
+      set_db.Members(key, &str_vec);
+
+      if (args.dontsort) {
+        str_vec = std::vector(str_vec.begin() + offset, str_vec.begin() + offset + count);
+      }
+    } else if (type == RedisType::kRedisZSet) {
+      auto zset_db = redis::ZSet(storage_, namespace_);
+      std::vector<MemberScore> member_scores;
+
+      if (args.dontsort) {
+        RangeRankSpec spec;
+        spec.start = offset;
+        spec.stop = offset + count - 1;
+        spec.reversed = args.desc;
+        zset_db.RangeByRank(key, spec, &member_scores, nullptr);
+
+        for (auto &member_score : member_scores) {
+          str_vec.emplace_back(member_score.member);
+        }
+      } else {
+        zset_db.GetAllMemberScores(key, &member_scores);
+
+        for (auto &member_score : member_scores) {
+          str_vec.emplace_back(member_score.member);
+        }
+      }
+    } else {
+      *res = SortResult::UNKNOWN_TYPE;
+      return s;
+    }
+  }
+
+  std::vector<RedisSortObject> sort_vec(str_vec.size());
+  for (size_t i = 0; i < str_vec.size(); ++i) {
+    sort_vec[i].obj = str_vec[i];
+  }
+
+  // Sort by BY, ALPHA, ASC/DESC
+  if (!args.dontsort) {
+    for (size_t i = 0; i < sort_vec.size(); ++i) {
+      std::string byval;
+      if (!args.sortby.empty()) {
+        byval = lookupKeyByPattern(args.sortby, str_vec[i]);
+        if (byval.empty()) continue;
+      } else {
+        byval = str_vec[i];
+      }
+
+      if (args.alpha) {
+        if (!args.sortby.empty()) {
+          sort_vec[i].v = byval;
+        }
+      } else {
+        try {
+          sort_vec[i].v = std::stod(byval);
+        } catch (const std::exception &e) {
+          *res = SortResult::DOUBLE_CONVERT_ERROR;
+          return rocksdb::Status::OK();
+        }
+      }
+    }
+
+    std::sort(sort_vec.begin(), sort_vec.end(),
+              [args](const RedisSortObject &a, const RedisSortObject &b) { return SortCompare(a, b, args); });
+
+    // Gets the element specified by Limit
+    if (offset != 0 || count != vectorlen) {
+      sort_vec = std::vector(sort_vec.begin() + offset, sort_vec.begin() + offset + count);
+    }
+  }
+
+  // Get the output
+  for (auto &elem : sort_vec) {
+    if (args.getpatterns.empty()) {
+      output_vec->emplace_back(elem.obj);
+    }
+    for (const std::string &pattern : args.getpatterns) {
+      std::string val = lookupKeyByPattern(pattern, elem.obj);
+      if (val.empty()) {
+        output_vec->emplace_back(redis::NilString(version));
+      } else {
+        output_vec->emplace_back(val);
+      }
+    }
+  }
```

Comment:
I don't think it's a good idea to generate output in `Database::Sort`, instead it should be in `Execute`.

`Database::Sort` can just focus on the algorithm and rocksdb operations.

Reply:
> Do subsequent store operations also need to be placed in Execute?

No. As I said, rocksdb read/write should be in `redis_db.cc`.

Reply:
This seems unfeasible. The code in the `output_vec` part is also necessary code in the store part and cannot be moved out separately.
At present, I have renamed `output_vec` -> `elems`, and removed the comment of output, so that semantically it can be considered that it is just a store operation.